### ISSUE TITLE
[Bug] PointRCNN training problem with much lower evaluation results and higher loss

### ISSUE THREAD

**Issue by Galaxy-ZRX** [open-mmlab/mmdetection3d#2123]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmdetection3d/issues) and [Discussions](https://github.com/open-mmlab/mmdetection3d/discussions) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://github.com/open-mmlab/mmdetection3d/blob/1.1/docs/en/notes/faq.md) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (dev)](https://github.com/open-mmlab/mmdetection3d/tree/dev) or [latest version (1.x)](https://github.com/open-mmlab/mmdetection3d/tree/dev-1.x).

### Task

I'm using the official example scripts/configs for the officially supported tasks/models/datasets.

### Branch

master branch https://github.com/open-mmlab/mmdetection3d

### Environment

sys.platform: linux
Python: 3.7.15 (default, Nov  7 2022, 22:00:21) [GCC 11.2.0]
CUDA available: True
GPU 0: Quadro RTX 8000
CUDA_HOME: /scratch/rz6u20/anaconda3/envs/mmdet3d_latest
NVCC: Cuda compilation tools, release 10.1, V10.1.24
GCC: gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-39)
PyTorch: 1.6.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.5.0 (Git Hash e2ac1fac44c5078ca927cb9b90e1b3066a0b2ed0)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.3
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 

TorchVision: 0.7.0
OpenCV: 4.6.0
MMCV: 1.5.3
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 10.1
MMDetection: 2.24.0
MMSegmentation: 0.25.0
MMDetection3D: 1.0.0rc5+962fc83
spconv2.0: False

### Reproduces the problem - code sample

I have been using mmdet3d to train PointRCNN from the beginning. I can train and evaluate the model but the results are much lower than the reported ones.

I modified the codes a little bit as below:

I added a coordinate shift parameter in kitti_dataset.py and loading.py, which is used to shift the loaded points, GT_bounding boxes and predicted bounding boxes. Relevant codes are shown below:

For the points:
`points[:, 0:3] += np.array(self.SHIFT_COOR, dtype=np.float32)` in [mmdet3d/datasets/pipelines/loading.py line 433](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/datasets/pipelines/loading.py#L433)

For the GT_boxes:
`gt_bboxes_3d.tensor[:, 0:3] += np.array(self.shift_coor, dtype=np.float32)` in [mmdet3d/datasets/kitti_datasets.py line 199](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/datasets/kitti_dataset.py#L199)
`sampled_gt_box[:, 2] += self.SHIFT_COOR_DB[2]` in [mmdet3d/datasets/pipelines/dbsampler.py line line 264](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/datasets/pipelines/dbsampler.py#L264)

For the predicted boxes:
`box_preds.tensor[:, 0:3] -= np.array(self.shift_coor, dtype=np.float32)` in [mmdet3de/datasets/kitti_datasets.py line 650](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/datasets/kitti_dataset.py#L650)

I am not sure if these modifications are ok or not, so I set the shift_coor value as [0.0, 0.0, 0.0] so that it will have no effect at all. But I got actually similar results. Therefore I just report the results with shift_coor value set as [0.0, 0.0, 1.6]. Before I train the model, I have validated that the sampled gt boxes are of similar values as the real gt boxes, which means this should not affect the dbsampler process. If I missed something here please point out without mercy ;)

As described above, I use 1*RTX8000 GPU with batch_size = 8, lr = 0.0005
I trained the model with config as: https://drive.google.com/file/d/1lqaTpNoh1jtwbemGV_fldc7mfb5uPayw/view?usp=sharing
And I got the log and evaluation results as: https://drive.google.com/file/d/1j0WeiFMsOsuNAn5D6SITVqFAvKDWABZ4/view?usp=sharing

In short, I got the evaluation results as below after 40 epochs:
2022-12-08 20:09:02,935 - mmdet - INFO - Saving checkpoint at 40 epochs
2022-12-08 20:32:16,812 - mmdet - INFO - 
Car AP@0.70, 0.70, 0.70:
bbox AP:72.9221, 54.3663, 48.0276
bev  AP:70.7341, 51.0141, 45.7713
3d   AP:57.7842, 38.5164, 33.8544
aos  AP:71.85, 53.05, 46.69
Car AP@0.70, 0.50, 0.50:
bbox AP:72.9221, 54.3663, 48.0276
bev  AP:78.2809, 62.4754, 55.7741
3d   AP:77.9438, 60.2450, 53.8742
aos  AP:71.85, 53.05, 46.69

I am not confident that it will end up with a close result to the reported ones since the total loss has been stable at around 1.4 and I got quite similar results for epoch 30 and 40. Could you please check if I have put anything wrong or should I just keep training? (Actually, I am letting the training continue now, and I will report the latest results once I get them.)

### Reproduces the problem - command or script

None

### Reproduces the problem - error message

None

**Comment by Galaxy-ZRX** [comment]
Update: results after 110 epochs:
```
2022-12-10 16:52:16,682 - mmdet - INFO - Epoch [110][900/928]	lr: 2.357e-04, eta: 5:52:55, time: 2.284, data_time: 0.010, memory: 18449, bbox_loss: 0.5832, semantic_loss: 0.1050, loss_cls: 0.3192, loss_bbox: 0.0610, loss_corner: 0.0278, loss: 1.0962, grad_norm: 1.0854
2022-12-10 16:53:20,398 - mmdet - INFO - Saving checkpoint at 110 epochs
2022-12-10 17:16:23,686 - mmdet - INFO - 
Car AP@0.70, 0.70, 0.70:
bbox AP:81.5227, 64.0027, 57.8342
bev  AP:76.9639, 59.9465, 55.5164
3d   AP:70.8566, 48.3033, 44.8050
aos  AP:80.69, 62.60, 56.41
Car AP@0.70, 0.50, 0.50:
bbox AP:81.5227, 64.0027, 57.8342
bev  AP:85.2150, 70.4637, 64.8907
3d   AP:84.6669, 68.7472, 63.0553
aos  AP:80.69, 62.60, 56.41
```

**Comment by Galaxy-ZRX** [comment]
Update: have also tried with SECOND in model zoo without any problem. The results after 10 epochs are:
```
2022-12-13 11:50:08,748 - mmdet - INFO - Saving checkpoint at 10 epochs
2022-12-13 11:53:39,726 - mmdet - INFO - 

----------- AP11 Results ------------

Car AP11@0.70, 0.70, 0.70:
bbox AP11:89.9018, 86.0105, 82.5927
bev  AP11:87.6454, 78.5815, 77.6583
3d   AP11:74.1530, 64.7638, 59.7841
aos  AP11:89.69, 85.21, 81.49
Car AP11@0.70, 0.50, 0.50:
bbox AP11:89.9018, 86.0105, 82.5927
bev  AP11:90.4033, 89.1014, 88.5015
3d   AP11:90.2808, 88.7431, 87.6196
aos  AP11:89.69, 85.21, 81.49
```

**Comment by VVsssssk** [comment]
Hi, thanks for your issue.
First, I think you can use tools to visualize some results to verify your modification, just like [demo. py](https://github.com/open-mmlab/mmdetection3d/tree/master/demo/pcd_demo.py) or [browse_dataset.py](https://github.com/open-mmlab/mmdetection3d/blob/master/tools/misc/browse_dataset.py). 
And then, about your modification, I don't know why we need to add a coordinate shift.  I think it could affect some transformation that relies on points and gt_box coordinates,  for example, `PointSample` or `ObjectRangeFilter`. So, I suggest you only add the shift of points and gt_boxes in [forward_train](https://github.com/open-mmlab/mmdetection3d/blob/5b8fb4bc60fda8ceed27c5185ba1841bc99e541a/mmdet3d/models/detectors/point_rcnn.py#L59)  and [simple_test](https://github.com/open-mmlab/mmdetection3d/blob/5b8fb4bc60fda8ceed27c5185ba1841bc99e541a/mmdet3d/models/detectors/point_rcnn.py#L110)  func to ensure the pipeline and evaluation were unaffected and try again. :+1:

**Comment by Galaxy-ZRX** [comment]
Before I undo the changes and only add the shift in forward and simple_test functions as you suggest, I am still worried that maybe this is not the key factor of these results as our comparisons. Could you give me some hints so that I can further test and validate? Thank you very much!
